pytorch_lightning/core/optimizer.py

```diff
@@ -133,17 +130,10 @@ def __optimizer_step(self, *args, closure: Optional[Callable] = None, profiler_n
         optimizer = self._optimizer
         model = trainer.get_model()
 
-        if trainer._device_type == DeviceType.TPU:
-            with trainer.profiler.profile(profiler_name):
-                xm.optimizer_step(optimizer, optimizer_args={'closure': closure, **kwargs})
-
-        elif trainer.amp_backend is not None:
-            trainer.precision_connector.backend.optimizer_step(trainer, optimizer, closure)
-
-        else:
-            with trainer.profiler.profile(profiler_name):
-                optimizer.step(closure=closure, *args, **kwargs)
-
+        with trainer.profiler.profile(profiler_name):
+            trainer.accelerator_backend.optimizer_step(*args, lambda_closure=closure, **kwargs)
+            
+        # TODO: Do we need this?
         accelerator_backend = trainer.accelerator_backend
```

Comment:
@awaelchli what about this rpc stuff? I feel like this should be handled by accelerator or plugin.

Reply:
yes this has to go to the rpc plugin. 
the ddp_plugin referred to here is the old one, not ours